tests/gummyroll_crud.ts

```diff
@@ -0,0 +1,358 @@
+      .add(allocGummyrollAccountIx)
+      .add(initGummyrollTx);
+    const initGummyRollTxId = await Gummyroll.provider.send(
+      tx,
+      [feePayerKeypair, merkleRollKeypair],
+      {
+        commitment: "confirmed",
+      }
+    );
+    assert(initGummyRollTxId, "Failed to initialize an empty Gummyroll");
+  });
+  describe("`Add` instruction", () => {
+    describe("having appended the first item", () => {
+      const firstTestMessage = "First test message";
+      beforeEach(async () => {
+        await appendMessage(firstTestMessage);
+      });
+      it("updates the root hash correctly", async () => {
+        const actualRoot = await getActualRoot();
+        const expectedRoot = recomputeRootByAddingLeafToTreeWithMessageAtIndex(
+          feePayerKeypair.publicKey,
+          firstTestMessage,
+          0
+        );
+        expect(expectedRoot.compare(actualRoot)).to.equal(
+          0,
+          "On-chain root hash does not equal expected hash"
+        );
+      });
+      describe("having appended the second item", () => {
+        const secondTestMessage = "Second test message";
+        beforeEach(async () => {
+          await appendMessage(secondTestMessage);
+        });
+        it("updates the root hash correctly", async () => {
+          const actualRoot = await getActualRoot();
+          recomputeRootByAddingLeafToTreeWithMessageAtIndex(
+            feePayerKeypair.publicKey,
+            firstTestMessage,
+            0
+          );
+          const expectedRoot =
+            recomputeRootByAddingLeafToTreeWithMessageAtIndex(
+              feePayerKeypair.publicKey,
+              secondTestMessage,
+              1
+            );
+          expect(expectedRoot.compare(actualRoot)).to.equal(
+            0,
+            "On-chain root hash does not equal expected hash"
+          );
+        });
+      });
+    });
+  });
+  describe("`Transfer` instruction", () => {
+    const message = "Message";
+    async function transferMessage(
+      newOwnerPubkey: PublicKey,
+      index: number,
+      config: { overrides?: { message?: string; signer?: Keypair } } = {}
+    ) {
+      const proofNodes = getProofOfLeaf(tree, index).map(({ node }) => node);
+      const signer = config.overrides?.signer;
+      const transferIx = GummyrollCrud.instruction.transfer(
+        Buffer.from(tree.root, 0, 32),
+        Buffer.from(config.overrides?.message ?? message),
+        proofNodes,
+        0,
+        {
+          accounts: {
+            gummyrollProgram: Gummyroll.programId,
+            merkleRoll: merkleRollKeypair.publicKey,
+            newOwner: newOwnerPubkey,
+            owner: feePayerKeypair.publicKey,
+          },
+          signers: [signer ?? feePayerKeypair],
+        }
+      );
+      const tx = new Transaction().add(transferIx);
+      await GummyrollCrud.provider.send(tx, [signer ?? feePayerKeypair], {
+        commitment: "confirmed",
+      });
+    }
+    beforeEach(async () => {
```

Comment:
Think we only want `appendMessage` to run once at beginning of entire test that way we test this on devnet without burning devnet sol